Iterable-iOS-SDK/IterableAPI.m

```diff
@@ -223,18 +223,39 @@ - (void)sendRequest:(NSURLRequest *)request onSuccess:(void (^)(NSDictionary *))
                                                                    NSURLResponse *response,
                                                                    NSError *error)
     {
-        if ([data length] > 0 && error == nil) {
-            error = nil;
-            id object = [NSJSONSerialization
-                         JSONObjectWithData:data
-                         options:0
-                         error:&error];
-            if(error) {
+        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+        NSInteger responseCode = httpResponse.statusCode;
+        
+        NSError *jsonError = nil;
+        id json = [NSJSONSerialization
+                     JSONObjectWithData:data
+                     options:0
+                     error:&jsonError];
+
+        if (responseCode == 401) {
+            if (onFailure != nil) onFailure(@"Invalid API Key", data);
+        }
+        else if (responseCode >= 400) {
+            NSString *errorMessage = @"Invalid Request";
```

Comment:
move 239-248 inside the `if (onFailure != nil)` on 250; not point in building the error message if it'll be unused?